### PR TITLE
Feature/movable legends

### PR DESCRIPTION
**Pull request by flibbertigibbet**

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
This updates the jquery-ui components and adds the draggable component. Also adds an example and modifies legends to allow them to bubble up interactivity events.

File: examples/tutorial-google-driving-2.html [hunk 1]
```diff
@@ -51,7 +51,6 @@
         .done(function(layers) {
             var subLayer = layers.getSubLayer(0);
             subLayer.setInteraction(true); // Interaction for that layer must be enabled
-            cdb.vis.Vis.addCursorInteraction(map, subLayer); // undo with removeCursorInteraction
```
Comment (CloudNiner) [inline examples/tutorial-google-driving-2.html]: Why was this removed, seems unrelated?

File: examples/draggable-legends.html [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+          $(".legend-selector a.selected").removeClass("selected");
+          $link.addClass("selected");
+
+          currentLegend = legends[type].legend;
+          legendSelector = '.cartodb-legend';
+        }
+
+        currentLegend.addTo(".legends");
+
+        $(legendSelector)
```
Comment (CloudNiner) [inline examples/draggable-legends.html]: Since `selectLegend` could be called multiple times, is it necessary to destroy the draggable before recreating it on the `$(legendSelector)` element?

File: src/cartodb.js [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
     var cdb = root.cdb = {};
 
-    cdb.VERSION = "3.15.9";
+    cdb.VERSION = "";
```
Comment (CloudNiner) [inline src/cartodb.js]: No issues with this being blank? @javisantana is there a particular version scheme the client would like to follow for this fork of cartodb.js?

File: src/geo/ui/legend.js [hunk 4]
```diff
@@ -117,18 +117,6 @@ cdb.geo.ui.Legend = cdb.core.View.extend({
 
   className: "cartodb-legend",
 
-  events: {
-    "dragstart":            "_stopPropagation",
```
Comment (CloudNiner) [inline src/geo/ui/legend.js]: I worry about this causing issues down the line. Do all of these need to be removed, or can we just remove a subset for what we need to make the draggable possible?

File: src/cartodb.js (hunk 3 above)
Comment (javisantana) [inline src/cartodb.js]: I'd keep it because it's pretty helpful when debugging, i'd do "3.15.9-bb" or something like that

**Comment by CloudNiner** [comment]
As described here, https://github.com/cartodb-org/cartodb/issues/19, since cartodb.js is the player or read-only mode, I think we'll want to put the code that properly positions a dragged legend into cartodb.js based on a custom x/y saved to the vis.json. But that's farther down the line...just something to keep in mind.

**Comment by CloudNiner** [comment]
+1 looks like a good start

File: examples/tutorial-google-driving-2.html (hunk 1 above)
Comment (flibbertigibbet) [inline examples/tutorial-google-driving-2.html]: See commit comment. Simply running `grunt` on the clean repo produces this change.

File: examples/draggable-legends.html (hunk 2 above)
Comment (flibbertigibbet) [inline examples/draggable-legends.html]: As it is, you can switch back to a previous legend and it will retain its placement. Destroying it would revert the placement I think.

File: src/cartodb.js (hunk 3 above)
Comment (flibbertigibbet) [inline src/cartodb.js]: Ran into some issues with the grunt commands becoming unhappy without the version defined; it would substitute an empty string for the version in the examples and break things.  Wasn't sure what a good approach would be for sub-versioning our fork; what would we do with the next change?

File: src/geo/ui/legend.js (hunk 4 above)
Comment (flibbertigibbet) [inline src/geo/ui/legend.js]: The only implication I've found for this is that within the editor interface (but not using cartodb.js directly) the map will listen to events propagated from the legend unless stopped, so it seems like me like the editor library should handle stopping propagation instead of simply killing everything here.

Several of these are related to drag; killing click, for instance, will prevent the drag start from being detected.

**Comment by flibbertigibbet** [comment]
Right, that's why this PR has no location-storing stuff in it; going to put that in the editor. Just needed to modify it to allow events to bubble up.

File: src/cartodb.js (hunk 3 above)
Comment (flibbertigibbet) [inline src/cartodb.js]: Oh I see, running `grunt` again with the version set in `package.json` and `bower.json` failed to reset the version in this file. I'll make that change manually then.

File: src/geo/ui/legend.js (hunk 4 above)
Comment (CloudNiner) [inline src/geo/ui/legend.js]: Sounds good.
